crate_universe/src/metadata.rs

```diff
@@ -124,6 +130,29 @@ impl Cargo {
         }
         Ok(full_version.clone().unwrap())
     }
+
+    pub fn use_sparse_registries_for_crates_io(&self) -> Result<bool> {
+        let full_version = self.full_version()?;
+        let version_str = full_version.split(' ').nth(1);
+        if let Some(version_str) = version_str {
+            let version = Version::parse(version_str).context("Failed to parse cargo version")?;
```

Comment:
Do you know if nightly releases return a clean semver value?

Reply:
They do, e.g. 1.69.0 parses as:
`Version { major: 1, minor: 69, patch: 0, pre: Prerelease("nightly") }`

Now, technically that means logic is _slightly_ buggy, because 1.68.0-nightly is not >= 1.68.0 - changed the logic to just compare the major and minor version.